internal/oci/runtime_oci.go

```diff
@@ -1065,6 +1065,17 @@ func (r *runtimeOCI) UpdateContainerStatus(ctx context.Context, c *Container) er
 
 		// Collect metric by container name
 		metrics.Instance().MetricContainersOOMCountTotalInc(c.Name())
+	} else if *c.state.ExitCode == 137 && r.config.CgroupManager().IsSystemd() {
+		// Due to a race between conmon reading cgroup files
+		// to find out whether OOM has happened, and systemd
+		// removing the cgroup, conmon might not detect an OOM.
+		// Systemd surely got it, but the unit is now removed thanks
+		// to "CollectMode: inactive-or-failed". The only remaining
```

Comment:
should we change the collect mode instead and make sure systemd cleans up the failed units on container rm?